pkg/scheduler/objects/application.go

```diff
@@ -97,6 +104,13 @@ func NewApplication(siApp *si.AddApplicationRequest, ugi security.UserGroup, eve
 	if time.Duration(0) == placeholderTimeout {
 		placeholderTimeout = defaultPlaceholderTimeout
 	}
+	gangSchedStyle := siApp.GetGangSchedulingStyle()
+	if gangSchedStyle != Soft && gangSchedStyle != Hard {
+		log.Logger().Info("Unknown gang scheduling style, using hard style as default",
+			zap.String("gang scheduling style", gangSchedStyle))
+		gangSchedStyle = Hard
+	}
```

Comment:
we should use `soft` as default

Reply:
Since we are already using Hard by default in our 0.10 release, if we change it to Soft, can break some workflows, not? I used Hard as default, to keep the same behaviour as before.